tasks/add-unique-granuleID/src/index.ts

```diff
@@ -0,0 +1,54 @@
+import Logger from '@cumulus/logger';
+import { Context } from 'aws-lambda';
+
+import { generateUniqueGranuleId } from '@cumulus/ingest/granule';
+import { CumulusMessage, CumulusRemoteMessage } from '@cumulus/types/message';
+import { CumulusMessageWithAssignedPayload, runCumulusTask } from '@cumulus/cumulus-message-adapter-js';
+
+import { HandlerEvent, HandlerOutput } from './types';
+
+const log = new Logger({ sender: '@cumulus/add-unique-granule-id' });
+
+/**
+ * If no producerId is assigned,
+ * assign uniqueIds to the granules in the payload,
+ * preserving the original ID in 'producerGranuleId'.
+ *
+ * @param event - input from the message adapter
+ * @returns the uri of the pan
+ */
+async function assignUniqueIds(event: HandlerEvent): Promise<HandlerOutput> {
+  const { granules } = event.input;
+
+  const output = {
+    granules: granules.map((granule) => {
```

Comment:
On second thought.... we could probably use mutation here given it's isolate and reduce memory footprint.

Reply:
I'm thinking I'm going to go with: https://github.com/nasa/cumulus/pull/3980, leaving up for review commentary for now.